drivers/quota/projectquota_supported.go

```diff
@@ -370,6 +371,20 @@ func setProjectID(targetPath string, projectID uint32) error {
 		return fmt.Errorf("failed to set projid for %s: %w", targetPath, errno)
 	}
 
+	// While we are setting the PROJINHERIT flag, it only applies to files
+	// and directories created after the fact.
+	// Any pre-existing subdirectories will still need to have the project
+	// ID set manually, so recurse onto them.
+	if dents, err := os.ReadDir(targetPath); err == nil {
```

Comment:
If we should do this, I’d expect to take advantage of existing `WalkDir`.

Reply:
Why? WalkDir seems to do more work we don't need here, this version seems to be much simpler than calling WalkDir() here.